packages/actor-rdf-resolve-hypermedia-sparql/lib/RdfSourceSparql.ts

```diff
@@ -0,0 +1,251 @@
+import {ActorHttp, IActionHttp, IActorHttpOutput} from "@comunica/bus-http";
+import {Bindings, BindingsStream} from "@comunica/bus-query-operation";
+import {ActionContext, Actor, IActorTest, Mediator} from "@comunica/core";
+import {namedNode, variable} from "@rdfjs/data-model";
+import {AsyncIterator, BufferedIterator} from "asynciterator";
+import {PromiseProxyIterator} from "asynciterator-promiseproxy";
+import * as RDF from "rdf-js";
+import {getTerms, getVariables, mapTerms} from "rdf-terms";
+import {Algebra, Factory, toSparql} from "sparqlalgebrajs";
+import {SparqlJsonParser} from "sparqljson-parse";
+
+export class RdfSourceSparql implements RDF.Source {
+
+  protected static readonly FACTORY: Factory = new Factory();
+
+  private readonly url: string;
+  private readonly context: ActionContext;
+  private readonly mediatorHttp: Mediator<Actor<IActionHttp, IActorTest, IActorHttpOutput>,
+    IActionHttp, IActorTest, IActorHttpOutput>;
+
+  constructor(url: string, context: ActionContext,
+              mediatorHttp: Mediator<Actor<IActionHttp, IActorTest, IActorHttpOutput>,
+                IActionHttp, IActorTest, IActorHttpOutput>) {
+    this.url = url;
+    this.context = context;
+    this.mediatorHttp = mediatorHttp;
+  }
+
+  /**
+   * Replace all blank nodes in a pattern with variables.
+   * If the pattern contains no blank nodes the original pattern gets returned.
+   * @param {RDF.BaseQuad} pattern A quad pattern.
+   * @return {RDF.BaseQuad} A quad pattern with no blank nodes.
+   */
+  public static replaceBlankNodes(pattern: RDF.BaseQuad): RDF.BaseQuad {
```

Comment:
Since it's not all that easy to find everything through the commits :sweat_smile: : Is this the same code as the other sparql actor that already exists (if yes why no reuse, if no nvm).

Reply:
Yes, it's exactly the same code. But I deliberately copy-pasted it, as the other SPARQL actor has been deprecated (and will remain unused). I intend to remove the old SPARQL actor completely in the next major version, so the abstraction would not be needed.